AlphaWallet/Transactions/Coordinators/SingleChainTransactionDataCoordinator.swift

```diff
@@ -140,7 +152,8 @@ class SingleChainTransactionDataCoordinator: Coordinator {
 
         let startBlock: Int
         let sortOrder: AlphaWalletService.SortOrder
-        if let newestCachedTransaction = storage.completedObjects.first {
+        //Do not use the transactions block number if it has been retrieved by erc20 event logs
+        if let newestCachedTransaction = storage.completedObjects.first, !newestCachedTransaction.isERC20Interaction {
```

Comment:
I think you'll have to do a loop of some sort (not efficient) or create a new property (like completedObjects, but add a filter for isERC20Interaction so this line is simply changed to (with a better name than `completedObjectsThatAreNotERC20Interaction` hopefully :) ):

```
if let newestCachedTransaction = storage.completedObjectsThatAreNotERC20Interaction.first, !newestCachedTransaction.isERC20Interaction {
```

Otherwise, if the last transaction in the database is `isERC20Interaction=true`, then it'll (wrongly) fetch the oldest transaction that it can (blockNumber 1)